### PR TITLE
adds image selection controls

### PR DESCRIPTION
**Pull request by danton267**
After the tiled image source is introduced, a few changes need to be made, to accommodate for data source buckets, see the comment below:

https://github.com/mlexchange/mlex_highres_segmentation/blob/c711cf042e1e919910371f6082f1b8af77d541cc/callbacks/image_viewer.py#L38-L45

as well as changing this line to work with the new data:
https://github.com/mlexchange/mlex_highres_segmentation/blob/c711cf042e1e919910371f6082f1b8af77d541cc/callbacks/image_viewer.py#L14

### CODE REVIEW

File: callbacks/image_viewer.py
Comment (hannahker) [inline callbacks/image_viewer.py:38]: It might be worth making this logical while we wait for the data set up from `tiled`. Correct me if I'm wrong, but this callback will only run on page load right now? 

What if in the "Data selection" dropdown someone selects based on a list of directory names (rather than specific file names). Each directory could have the set of `.tiff` files that make up a given dataset. Then once someone makes that selection, this callback runs based on the contents of the directory that they've selected.

**Review by hannahker — CHANGES_REQUESTED** [review]
Nice work! Added a couple of minor comments to be addressed before merging.

File: components/image_viewer.py
Comment (danton267) [reply components/image_viewer.py:20]: With overlay hidden it would, IMO, be difficult to notice something is happening. I'd prefer to change color of it to make it more obious (right now it's gray), keep it at .15-.35, which would still give a good opacity and udnerstanding to the user that their action just changed the image

File: components/image_viewer.py
Comment (danton267) [reply components/image_viewer.py:29]: I changed its location.

I have tried various resolutions, and the image looks okay for "standard" non-mobile resolutions on my end? We could force it to be responsive to always fit the screen, and prevent scroll-behavior (which only happens on mobile-like res)

File: callbacks/image_viewer.py
Comment (danton267) [reply callbacks/image_viewer.py:38]: The callback is trigger every time someone changes the dropdown with the img-source, yes.

However, it is ready for when that will include folders, not these file names. We simply add a dcc.Store which will be filled with the file names, when the data source is selected, and we will then use it here, to trigger this callback and use that data to plot the images.

Comment (danton267) [reply callbacks/image_viewer.py]: See c55295b commit in the PR, i have addressed the multi-project / folder + image selections

File: utils/data_utils.py
Comment (hannahker) [inline utils/data_utils.py:6]: Just need to add the following otherwise this will throw an error if the `data` dir doesn't exist: 

```
    if not os.path.exists(dir):
        os.mkdir(dir) 
```

File: utils/data_utils.py
Comment (hannahker) [inline utils/data_utils.py:23]: Very nice 🌟

**Comment by hannahker** [comment]
Nice work @danton267! See the one new comment about directory creation. This can be merged once that is addressed and once we've resolved the directory structuring issue on deploy.

File: utils/data_utils.py
Comment (tamidodo) [reply utils/data_utils.py:6]: I don't think this is actually the problem we're hitting though - the logs say it can't find "data/None" and I think it's because of this chain of events:
- DATA_OPTIONS = data_utils.get_data_project_names() is at the top of the control_bar.py file so it only runs once, on app deployment. The get_data_project_names() function works just fine, the data folder always exists because we created one in our code base to begin with. However, there are no images or project names in that data folder so we get to the next block of code
- dmc.Select(
      id="project-name-src",
      data=DATA_OPTIONS,
      value=DATA_OPTIONS[0] if DATA_OPTIONS else None,
      placeholder="Select an image to view...",
  ),
- Which means that the project-name-src value is None and then the update_slider_values() callback in image_viewer.py tries to run the data_utils.get_tiff_files(project_name) line but the project_name is None leading to the log error we see of FileNotFoundError: [Errno 2] No such file or directory: 'data/None'

Comment (tamidodo) [reply utils/data_utils.py:6]: So the problem is that the DEV_download_google_sample_data() function getting run in app.py is not guaranteed to be the first thing that happens. We need to move it to a predeploy script so that on app's first load, there is already that data there.

**Comment by tamidodo** [comment]
Ok, fixed it, it's working now

**Review by tamidodo — APPROVED** [review]

**Review by hannahker — APPROVED** [review]
Nice fix! This LGTM 💃

**Pull request merged**
